### PR TITLE
Fix formatting of TypeSpec KDoc

### LINKED ISSUE #563: KDoc formatting for types with properties is off
KDoc formatting for constructor parameter/properties is off.

Example:

```kotlin
import com.squareup.kotlinpoet.FileSpec
import com.squareup.kotlinpoet.FunSpec
import com.squareup.kotlinpoet.INT
import com.squareup.kotlinpoet.ParameterSpec
import com.squareup.kotlinpoet.PropertySpec
import com.squareup.kotlinpoet.TypeSpec

fun main(args: Array<String>) {
  val typeSpec = TypeSpec.classBuilder("MyType")
      .addKdoc("This is a thing for stuff.")
      .addProperty(
          PropertySpec.builder("first", INT)
              .initializer("first")
              .build()
      )
      .addProperty(
          PropertySpec.builder("second", INT)
              .initializer("second")
              .build()
      )
      .addProperty(PropertySpec.builder("third", INT)
          .initializer("third")
          .build()
      )
      .primaryConstructor(
          FunSpec.constructorBuilder()
              .addKdoc("Construct a thing!")
              .addParameter(ParameterSpec.builder("first", INT).addKdoc("the first thing").build())
              .addParameter(ParameterSpec.builder("second", INT).addKdoc("the second thing").build())
              .addParameter(ParameterSpec.builder("third", INT).addKdoc("the third thing").build())
              .build()
      ).build()

  FileSpec.get("", typeSpec).writeTo(System.out)
}


```

Output

```kotlin
import kotlin.Int

/**
 * This is a thing for stuff.@param first the first thing@param second the second thing@param third
        the third thing */
class MyType(
    val first: Int,
    val second: Int,
    val third: Int
)
```

Expected - something like:

```kotlin
import kotlin.Int

/**
  * This is a thing for stuff.
  *
  * @param first the first thing
  * @param second the second thing
  * @param third the third thing 
  */
class MyType(
    val first: Int,
    val second: Int,
    val third: Int
)
```

### PR DESCRIPTION
**Pull request by Egorand**
Fixes #563

### CODE REVIEW

File: src/main/java/com/squareup/kotlinpoet/Util.kt [hunk 1]
```diff
@@ -156,6 +156,19 @@ internal val String.isKeyword get() = this in KEYWORDS
 
 internal val String.isName get() = split("\\.").none { it.isKeyword }
 
+internal fun CodeBlock.ensureEndsWithNewLine() = if (isEmpty()) this else with(toBuilder()) {
+  val lastFormatPart = formatParts[formatParts.size - 1]
```
Comment (shaishavgandhi05) [inline src/main/java/com/squareup/kotlinpoet/Util.kt]: Can also do `formatParts.last()`

File: src/main/java/com/squareup/kotlinpoet/Util.kt [hunk 2]
```diff
@@ -156,6 +156,19 @@ internal val String.isKeyword get() = this in KEYWORDS
...
+  if (CodeBlock.isPlaceholder(lastFormatPart)) {
+    val lastArg = args[args.size - 1]
```
Comment (shaishavgandhi05) [inline src/main/java/com/squareup/kotlinpoet/Util.kt]: `args.last()`

File: src/test/java/com/squareup/kotlinpoet/TypeSpecTest.kt [hunk 3]
```diff
@@ -3568,6 +3569,48 @@ class TypeSpecTest {
...
+            .initializer("first")
+            .build())
+        .addProperty(PropertySpec.builder("second", INT)
+            .initializer("second")
+            .build())
+        .addProperty(PropertySpec.builder("third", INT)
+            .initializer("third")
+            .build())
+        .primaryConstructor(FunSpec.constructorBuilder()
+            .addKdoc("Construct a thing!")
```
Comment (swankjesse) [inline src/test/java/com/squareup/kotlinpoet/TypeSpecTest.kt]: Losing this doc is annoying.

Comment (swankjesse) [reply src/test/java/com/squareup/kotlinpoet/TypeSpecTest.kt]: Should we coalesce it? Not fold when there's constructor docs?

Comment (Egorand) [reply src/test/java/com/squareup/kotlinpoet/TypeSpecTest.kt]: There might be a special tag for constructor docs, I'll research

**Pull request merged**